index.js

```diff
@@ -11,7 +11,22 @@ const queryToSheet = require('./lib/query-to-sheet');
       auth: await googleAuth(),
     });
 
-    await Promise.all(queries.map(queryToSheet(googleSheets)));
+    // Function to process batches of queries
+    const processBatch = async (batch) => {
+      await Promise.all(batch.map(queryToSheet(googleSheets)));
+    };
+
+    // Split queries into batches of 50
+    // API says 60 a minute but still beem to be getting throttled
+    const batchSize = 50;
+    for (let i = 0; i < queries.length; i += batchSize) {
```

Comment:
Perhaps it might be simpler to have a function that creates a `batches` array, that you can then simply iterate over. For example (this is pseudocode -- haven't tested it):

```
const createBatches = (queries, batchSize) => {
  const batches = [];
  while (queries.length > 0) {
    batches.push(queries.splice(0, batchSize));
  }
  return batches;
};
```